### PR TITLE
fix(sharePanel): [FOR-426] hide public link on sharing close confirmation pop up

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
Check if sharePanel close confirmation pop up is open and if it is we do not show the public link and its button.

## Issue ticket number and link
FOR-426 : https://jira.support-ent.fr/browse/FOR-426

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts [hunk 1]
```diff
@@ -766,6 +767,12 @@ export const formsListController = ng.controller('FormsListController', ['$scope
         vm.limitTable += vm.tableSize;
     };
 
+    vm.isCloseConfirmationOpen = () : boolean => {
+        let elem = document.getElementsByTagName('share-panel')[0].firstElementChild;
+        let sharePanelScope = angular.element(elem).scope();
+        return sharePanelScope.display.showCloseConfirmation;
```
Comment (sjollois) [inline formulaire/src/main/resources/public/ts/controllers/forms-list.ts:773]: rajouter un if si sharePanelScope et sharePanelScope.display existent

Comment (JinDouang) [reply formulaire/src/main/resources/public/ts/controllers/forms-list.ts:773]: Yes, tu peux risquer un cannot read property of undefined (si display est undefined ou son parent sharePanelScope)

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts [hunk 2]
```diff
@@ -766,6 +767,12 @@ export const formsListController = ng.controller('FormsListController', ['$scope
         vm.limitTable += vm.tableSize;
     };
 
+    vm.isCloseConfirmationOpen = () : boolean => {
+        let elem = document.getElementsByTagName('share-panel')[0].firstElementChild;
+        let sharePanelScope = angular.element(elem).scope();
```
Comment (JinDouang) [inline formulaire/src/main/resources/public/ts/controllers/forms-list.ts:772]: Tu peux le typer avec ": IScope"
```suggestion
        let sharePanelScope: IScope = angular.element(elem).scope();
```

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts [hunk 3]
```diff
@@ -766,6 +767,12 @@ export const formsListController = ng.controller('FormsListController', ['$scope
         vm.limitTable += vm.tableSize;
     };
 
+    vm.isCloseConfirmationOpen = () : boolean => {
+        let elem = document.getElementsByTagName('share-panel')[0].firstElementChild;
```
Comment (JinDouang) [inline formulaire/src/main/resources/public/ts/controllers/forms-list.ts:771]: typage

```suggestion
        let elem: Element = document.getElementsByTagName('share-panel')[0].firstElementChild;
```

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts (hunk 2 above)
Comment (qperie) [reply formulaire/src/main/resources/public/ts/controllers/forms-list.ts]: Si je type IScope il reconnait pas le '.display' que je cherche après

File: formulaire/src/main/resources/public/ts/controllers/forms-list.ts (hunk 3 above)
Comment (qperie) [reply formulaire/src/main/resources/public/ts/controllers/forms-list.ts]: Le Element importé de l'entcore ne correspond pas au type Element de mon objet

Comment (sjollois) [reply formulaire/src/main/resources/public/ts/controllers/forms-list.ts]: tu as essayer Un autre import de Element ?

**Pull request merged**